### PR TITLE
feat: move runtime from egg

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Affected core subsystem(s)

context, middleware

##### Description of change

Ref https://github.com/eggjs/egg/issues/301

### CODE REVIEW

File: app/middleware/instrument.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+const ms = require('ms');
+const RUNTIME = require('../../lib/constant').RUNTIME;
+
+module.exports = options => {
+  return function* (next) {
+    if (options.enableConsole !== true) return;
```
Comment (atian25) [inline app/middleware/instrument.js]: 这里应该是 yield  next 而不是 return 吧？

Comment (popomore) [reply app/middleware/instrument.js]: return 就不会再进来了

Comment (fengmk2) [reply app/middleware/instrument.js]: 放在上一层，不是中间件代码里面

File: app/middleware/instrument.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    yield next;
+
+    const runtimes = [];
+    for (const { event, duration } of this[RUNTIME].entries()) {
+      runtimes.push(`${event}: ${ms(duration)}`);
+    }
+    this.logger.info(`status ${this.status} (${runtimes.join(', ')})`);
```
Comment (fengmk2) [inline app/middleware/instrument.js:18]: runtime是干嘛的？为什么要打印两次？没明白为何要加一个中间件

Comment (popomore) [reply app/middleware/instrument.js:18]: 原来的逻辑，统计一个请求所有 event 的耗时

Comment (popomore) [reply app/middleware/instrument.js:18]: @fengmk2 看看原来的 https://github.com/eggjs/egg-development/pull/5

**Comment by popomore** [comment]
依赖 https://github.com/eggjs/egg-development/pull/5

**Comment by popomore** [comment]
好了

**Review by fengmk2 — APPROVED** [review]

**Comment by popomore** [comment]
+ egg-instrument@1.1.0

**Pull request merged**